File: backend/src/main/java/com/lifelink/notification/entity/Notification.java
package com.lifelink.notification.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.lifelink.common.typehandler.JsonbStringTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName(value = "notifications", autoResultMap = true)
public class Notification {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long receiverUserId;
    private Long actorUserId;
    private String notificationType;
    private String title;
    private String content;
    private String relatedType;
    private Long relatedId;
    private Long relationshipId;
    private String readStatus;
    private String status;

    @TableField(typeHandler = JsonbStringTypeHandler.class)
    private String metadata;

    private LocalDateTime createdAt;
    private LocalDateTime readAt;
}

File: web/src/api/notification.ts
import { ApiResult, request } from './request';

export type NotificationReadStatus = 'UNREAD' | 'READ';

export interface NotificationItem {
  id: number;
  receiverUserId: number;
  actorUserId?: number | null;
  actorUsername?: string | null;
  actorAvatarUrl?: string | null;
  notificationType: string;
  title: string;
  content?: string | null;
  relatedType?: string | null;
  relatedId?: number | null;
  relationshipId?: number | null;
  relationshipName?: string | null;
  readStatus: NotificationReadStatus;
  status: string;
  metadata: Record<string, unknown>;
  createdAt: string;
  readAt?: string | null;
}

export interface NotificationUnreadCount {
  count: number;
}

export interface NotificationListParams {
  readStatus?: NotificationReadStatus;
  notificationType?: string;
  page?: number;
  size?: number;
}

export function getNotifications(params: NotificationListParams = {}) {
  return request.get<ApiResult<NotificationItem[]>>('/api/notifications', { params });
}

export function getNotificationUnreadCount() {
  return request.get<ApiResult<NotificationUnreadCount>>('/api/notifications/unread-count');
}

export function markNotificationAsRead(id: number) {
  return request.patch<ApiResult<void>>(`/api/notifications/${id}/read`);
}

export function markAllNotificationsAsRead() {
  return request.patch<ApiResult<void>>('/api/notifications/read-all');
}

export function deleteNotification(id: number) {
  return request.delete<ApiResult<void>>(`/api/notifications/${id}`);
}
